storage/client_raft_test.go

```diff
@@ -239,7 +241,9 @@ func TestFailedReplicaChange(t *testing.T) {
 	}()
 	storage.TestingCommandFilter = func(method string, args proto.Request, reply proto.Response) bool {
 		if method == proto.EndTransaction && args.(*proto.EndTransactionRequest).Commit == true {
-			reply.Header().SetGoError(util.Errorf("boom"))
+			oldErr := unsafe.Pointer(&reply.Header().Error)
+			newErr := proto.Error{Message: "boom"}
+			atomic.SwapPointer((*unsafe.Pointer)(oldErr), unsafe.Pointer(&newErr))
```

Comment:
Yeah, except that that doesn't actually store the new error.. I had to play around with it a fair bit to end up with the one above that actually worked.